vm/devices/pci/vpci_relay/src/linux_mmio.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! MMIO access types for Linux environments.
+
+use crate::CreateMemoryAccess;
+use anyhow::Context as _;
+use hcl::ioctl::MshvHvcall;
+use std::sync::Arc;
+use vpci_client::MemoryAccess;
+
+/// Accesses MMIO space directly via `/dev/mem`.
+pub struct DirectMmio(fs_err::File);
+
+impl DirectMmio {
+    /// Opens `/dev/mem` for MMIO access.
+    pub fn new() -> anyhow::Result<Self> {
+        let dev_mem = fs_err::OpenOptions::new()
+            .read(true)
+            .write(true)
+            .open("/dev/mem")
```

Comment:
we've talked about removing other `/dev/mem/` instances in the codebase, is there some way to remove this in the future, or should we just expect to use mmio hypercalls instead and remove this whole path?

Reply:
MMIO hypercalls aren't always supported, for whatever reason.

I think /dev/mem actually makes the most sense here, closest to its semi-supported use, which is to poke around with device registers that are not exposed properly through a driver.